angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
Your surprise is caused by the fact that filters weren't designed to be used from JS code. They were created for Angular's extensions of HTML. Their usage looks like this:
```html
    <tr ng-repeat="friend in friends | orderBy:'-age'">
      <td>{{friend.name}}</td>
      <td>{{friend.phone}}</td>
      <td>{{friend.age}}</td>
    </tr>
```
That's why I don't really understand the motivation behind the PRs like this. However, it's not the first one. I suppose people have their reasons why they want these changes.